drafts/too-many-tests.md

```diff
@@ -21,19 +23,56 @@ public void CallingFunctionAbcDoesStuff()
+
+This just leads to brittle tests that change whenever anyone does any refactoring!
+
+For years, companies and developers have talked about how great TDD is and sometimes about their struggles with it. The simple truth is though, that many developers have no idea what it really is! 
+
+I first became interested in tests after working on a project which wasn't as successful as we all hoped. Despite it being greenfield development, parts of the code quickly became very complicated as they were poorly structured. We spent days (literally) testing and retesting as bugs were fixed and then new ones inadvertently introduced.
+
+I remember the excitement of realising I could test classes by injecting the dependencies as interfaces. Suddenly it became easy to have loosely-coupled testable code. As more and more people started doing this, code quality improved, functionality became more reliable, and I no longer had to spend large amounts of time doing manual testing.
+
+However, in spite of the articles I read about about TDD, I found I could never really do it. Occassionally I would start with the tests when I needed to write an algorithm, but generally I found it easier to write the tests afterwards to confirm my code did what I expected it to do.
+
+Over time I became convinced there was something wrong. Tests and DI had given me some benefits, but it wasn't enough. Too many of the tests I wrote seemed worthless.
+
+## TDD Revisited
+
+What does TDD look like?
+
+- Write a failing test
+- Write code to make the test pass
+- Refactor
+
+Sounds simple. But when you are writing tests for each class it quickly becomes frustrating. Plus, the tests don't drive the design as you've already designed it in your head.
+
+The problem is that most developers, myself included, tend to think of a class as being a unit. Any tests that you write at that level will only test part of the problem you are trying to solve. And when you come back to them 6 months down the line, the chances are you're not going to understand what they do.
+
+Imagine a requirement to manage a basket on an e-commerce website. It might look like this:
+
+```
+Given I have an empty basket
+When I add an item to the basket
+Then my basket will contain the added item
+```
+
+In order to meet this requirement you will need a module for managing baskets. This might be a microservice, or just one endpoint among many on a webservice. Regardless of how it's implemented, the observable outcome will be the same. Adding an item to the basket will cause the basket to contain that item.
```

Comment:
In order to meet this requirement, you will need a module for managing baskets. This might be a microservice or just one endpoint among many on a web service. Regardless of how it's implemented, the observable outcome will be the same. Adding an item to the basket will cause the basket to contain that item.